### DISCUSSION TITLE
Converter with MarkupExtension

### DISCUSSION THREAD

**Discussion by agnauck** [AvaloniaUI/Avalonia#14421] [Q&A] [answered]
Is it possible to create converters with MarkupExtensions like here in WPF currently?
https://gist.github.com/stevegreatrex/3835815

I am looking to have a converter like:
`<TextBlock Text="{Binding IsActive, Converter={converters:ConditionalMarkupConverter TrueValue='Foo', FalseValue='Bar'}}" />`

**Comment by timunie** [comment]
should be, yes. Just add ProvideValue. #4765

**Comment by gehongyan** [comment, marked as answer]
Sure, it works. The code gist provided can be used in Avalonia directly if you disable the nullable feature.

```xaml
<!-- In Resources, defines a True value: <system:Boolean x:Key="True">True</system:Boolean> -->
<CheckBox Name="WhenCheckBox" />
<TextBlock Text="{Binding #WhenCheckBox.IsChecked,
                          Converter={localClient:When IsEqualTo={StaticResource True},
                              Return=Checked!,
                              Else=Unchecked...}}" />
```

![image](https://github.com/AvaloniaUI/Avalonia/assets/21241496/1b9d281c-beed-4818-98d7-50204f0388ee)
![image](https://github.com/AvaloniaUI/Avalonia/assets/21241496/72ee99ac-5869-4435-8ebf-346c1d61a20a)

**Comment by agnauck** [comment]
ok, I need to try again then.
I did not run it in Rider because there was no intellisense on the props and Rider reported errors in Xaml on the custom properties.

**Reply by agnauck** [reply]
looks like my issue was related to the nullable feature,
.
Thanks 👍
